File: packages/ios/RNEmbraceOTLPModule/RNEmbraceOTLPModule.m
#import <React/RCTBridgeModule.h>

@interface RCT_EXTERN_MODULE(RNEmbraceOTLPModule, NSObject)

RCT_EXTERN_METHOD(setCustomExporter:(NSString *)endpoint
                  header:(NSString *)header
                  token:(NSString *)token
                  resolver:(RCTPromiseResolveBlock)resolve
                  rejecter:(RCTPromiseRejectBlock)reject)

+ (BOOL)requiresMainQueueSetup
{
    return NO;
}

@end

File: packages/ios/RNEmbraceOTLPModule/RNEmbraceOTLPModule.swift
import Foundation

@objc(RNEmbraceOTLPModule)
class RNEmbraceOTLPModule: NSObject {
    @objc(setCustomExporter:header:token:resolver:rejecter:)
    func setCustomExporter (_ endpoint: String,
                            header: String,
                            token: String,
                            resolver resolve: RCTPromiseResolveBlock,
                            rejecter reject: RCTPromiseRejectBlock) {
        
        print("Native Module: RNEMbraceOTLP.setCustomExporter")

        resolve(true)
    }
}
